src/ralph/backends/http/async_lrs.py

```diff
@@ -150,10 +151,10 @@ def read(  # pylint: disable=too-many-arguments
             fragment="",
         ).geturl()
 
-        def fetch_statements(params: dict):
+        async def fetch_statements(params: dict):
             while True:
```

Comment:
:bulb: performance optimization ideas:

- instead of an infinite loop maybe we should make this function recursive
- to send a new concurrent request ASAP (before yielding all response statements), parsing the `more` link should be done first!

Reply:
IMO, recursivity can be [quite slow](https://stackoverflow.com/questions/13543019/why-is-recursion-in-python-so-slow) in terms of performance, and it makes the code more complicated to understand

Agree with the second idea though

Reply:
I didn't know about recursivity performance cost, thanks. In this case, it would be interesting to measure response time for a large batch of (paginated) statements: i. before, ii. after introducing recursivity, and iii. after moving the next concurrent request on top.

Reply:
Not sure I agree with the concurrent call before yielding. I understand the benefit but at the same time, we shouldn't be making non-necessary API calls (for the planet, the server, the client, and if a LRS user is rate-limited). This is the purpose of paginating the results in the first place.

Reply:
It was decided to implement a `greedy` option (default: False) which behaves as suggested by Julien. This allows for improved performance in the case of a client performing slow tasks upon receiving the statements (on the same time scale as the API calls)